File: Javascript/SelectionSort.js
function findMin (array) {
    let min = 0
    for (let counter = 1; counter < array.length; counter += 1) {
        if (array[counter] < array[min]) min = counter
    }
    return min
}

function selectionSort (array) {
    for (let counter = 0; counter < array.length; counter += 1) {
        let minLocation = findMin(array.slice(counter)) + counter
        if (minLocation !== counter) {
            let lastItem = array[counter]
            array[counter] = array[minLocation]
            array[minLocation] = lastItem
        }
    }
    return array
}

File: Python/SelectionSort.py
def findMin(array):
    min = 0
    for counter in range(1, len(array)):
        if array[counter] < array[min]:
            min = counter
    return min


def selectionSort(array):
    for counter in range(len(array)):
        minLocation = findMin(array[counter:]) + counter
        if minLocation != counter:
            lastItem = array[counter]
            array[counter] = array[minLocation]
            array[minLocation] = lastItem
    return array
